% Template for PLoS
% Version 3.6 Aug 2022
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended 
% to minimize problems and delays during our production 
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % % 
%
% Once your paper is accepted for publication, 
% PLEASE REMOVE ALL TRACKED CHANGES in this file 
% and leave only the final text of your manuscript. 
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.  For example, x$^2$ is incorrect; this should be formatted as $x^2$ (or $\mathrm{x}^2$ if the romanized font is desired).
%
% Do not include text that is not math in the math environment. For example, CO2 should be written as CO\textsubscript{2} instead of CO$_2$.
%
% Please add line breaks to long display equations when possible in order to fit size of the column. 
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.  For example, change "[U(D,E,\gamma)]^2" to "{[U(D,E,\gamma)]}^2". 
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % % 
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
%\documentclass[10pt]{article}
%\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

\usepackage[paper=a4paper,
  inner=2.5cm,
  outer=3.8cm,
  bindingoffset=.5cm,
  top=1.5cm,
  bottom=1.5cm]{geometry}


% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}
\usepackage{physics}
\newcommand{\pdvn}[3]{\frac{\partial^{#1} {#2}}{\partial {#3}^{#1}}}
\newcommand{\Underbrace}[2]{{\underbrace{#2}_{#1}}}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
% \usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage[nopatch=eqnum]{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}


% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
\usepackage{setspace} 
\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}
  
% Use the PLoS provided BiBTeX style
% \bibliographystyle{plos2015}
% Remove brackets from numbering in List of References

\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother




% Header and Footer with logo

\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
\usepackage[bibstyle=numeric,style=authoryear-icomp,maxbibnames=10,maxcitenames=2,uniquelist=false, backend=bibtex,url=false, sorting=none]{biblatex}
\usepackage{float}



\addbibresource{manual.bib}
\AtEveryBibitem{\clearfield{note}}



%\addbibresource{Mendeley.bib}
%\addbibresource{manual.bib}
%\bibliography{references}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

%% END MACROS SECTION

\setlength{\parskip}{\baselineskip}%
\setlength{\parindent}{0pt}%
\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Effects of multistability, absorbing boundaries and growth on Turing pattern formation} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Martina Oliver Huidobro \textsuperscript{1},
Robert G. Endres \textsuperscript{1}


\bigskip
\textbf{1} Department of Life Sciences and Centre for Integrative Systems Biology and Bioinformatics, Imperial College, London, United Kingdom
\\
\bigskip

% Insert additional author notes using the symbols described below. Insert symbol callouts after author names as necessary.
% 
% Remove or comment out the author notes below if they aren't used.
%
% Primary Equal Contribution Note
%\Yinyang These authors contributed equally to this work.

% Additional Equal Contribution Note
% Also use this double-dagger symbol for special authorship notes, such as senior authorship.
%\ddag These authors also contributed equally to this work.

% Current address notes
%\textcurrency Current Address: Life Sciences Department, Imperial College London, London, United Kingdom% change symbol to "\textcurrency a" if more than one current address note
% \textcurrency b Insert second current address 
% \textcurrency c Insert third current address

% Deceased author note
%\dag Deceased

% Group/Consortium Author Note
%\textpilcrow Membership list can be found in the Acknowledgments section.

% Use the asterisk to denote corresponding authorship and provide email address in note below.

* r.endres@imperial.ac.uk

\end{flushleft}
% Please keep the abstract below 300 words

\section*{Abstract}
Turing patterns are a fundamental concept in developmental biology, describing how homogeneous tissues develop into self-organized spatial patterns. However, the classical Turing mechanism, which relies on linear stability analysis, often fails to capture the complexities of real biological systems, such as multistability, non-linearities, growth, and boundary conditions. Here, we explore the impact of these factors on Turing pattern formation, contrasting linear stability analysis with numerical simulations based on a simple reaction-diffusion model, motivated by synthetic gene regulatory pathways. We demonstrate how non-linearities introduce multistability, leading to unexpected pattern outcomes not predicted by traditional Turing theory. Additionally, the study examines how growth and realistic boundary conditions influence pattern robustness, revealing that different growth regimes and boundary conditions can either disrupt or stabilize pattern formation. Our findings are critical for understanding pattern formation in both natural and synthetic biological systems, providing insights into engineering robust patterns for applications in synthetic biology.


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
\section*{Author summary}
During development, tissues self-organise to go from a single cell to a structured organism. In this process, simple chemical reactions lead to the emergence of the intricate designs we see in nature, like the stripes on a zebra or the labyrinths on a brain cortex. Although multiple theories have been proposed to model this phenomenon, one of the most simple and popular ones was introduced in the 1950s by the mathematician Alan Turing. However, his theory oversimplifies the biological conditions and ignores properties such as non-linearities, boundary effects or growth in the tissue. In this work, we used a combination of mathematical models and computer simulations to investigate how these real-world factors influence pattern formation. Our findings show that when we account for these realistic effects, the patterns that emerge can be very different from what Turing's theory would predict. Thus, this work may help us better understand the laws behind pattern formation and could have practical applications in tissue engineering for medical or environmental applications. 

\linenumbers

\section*{Introduction}

How biology produces robust patterns in space and time is still largely an open question~\parencite{scholes2017three}.
In 1952, Alan Turing proposed a mechanism referred to as Turing patterns or diffusion-driven instabilities, which explains how a homogeneous tissue results in self-organized spatial repetitive patterns of the network’s molecules ~\parencite{Turing1952, Gierer1972}.
However, this proposed mechanism is far from real biological complexity, and suffers from fine tuning, meaning only a small subset of constrained parameters can generate these spatial patterns~\parencite{maini2012turing, Scholes2019}.
Additionally, the diffusion-driven instability is based on linear stability analysis that often does not address realistic phenomena such as multistability, growth, exotic boundary conditions, and non-linearities~\parencite{Krause2023}.
A way forward to understanding the relationship between theoretical Turing patterns and real biological patterns is to engineer these reaction-diffusion networks in bacterial colonies using synthetic biology~\parencite{Sekine2018, Karig2018}.
This would allow us to better understand the role of Turing patterns in biological pattern formation, as well as to engineer synthetic patterns for industrial applications~\parencite{cao2017programmable, tan2018polyamide,din2020interfacing}.
This approach was recently extended and implemented in~\cite{Oliver2023} where a Turing gene circuit was introduced in growing \textit{E.coli} colonies producing fluorescent periodic spatial patterns. Fig~\ref{fig1}A,B shows a final snapshot of a bacterial colony with periodic patterns in the GFP channel and the mean fluorescence in a cross section of the colony.
Understanding how patterns arise in this biological system is crucial for engineering synthetic patterning and to further understand the mechanisms of pattern formation in biology.

% Place figure captions after the first paragraph in which they are cited.
\begin{figure}[H]
    \includegraphics[width=1\textwidth]{figures/biological_example}

    \caption{{\bf Synthetic Turing patterns from experiments and theory.}
        \textbf{(A)} Last frame of a time lapse of a bacterial colony grown with a synthetic 6-node Turing gene circuit producing periodic spatial patterns in two-dimensions (scale bar: 1mm). Only the GFP channel is shown (\cite{Oliver2023}). \textbf{(B)} Mean fluorescence radial profile of the colony along the highlighted wedge with azimuthal averaging (yellow triangle). \textbf{(C)} Two-node network based on Turing’s original paper \cite{Turing1952} with slow diffusing activator $A$ in green and fast diffusor inhibitor $I$ in pink. \textbf{(D)} Kymograph of numerical solution for $A$ in 1D of the Turing circuit shown in C (corresponding to activator $A$). Domain is growing with absorbing boundaries. Plotted as a timeseries of molecule $A$ (nM). The $Y$ axis is time (h), $X$ axis is space (mm) and concentration is shown by color with blue being high concentration and red being low concentration. A periodic pattern appears as the tissue grows. \textbf{(E)} Final snapshot of solution in D shows a periodic pattern. The $Y$ axis is concentration (nM) of $A$ (left, blue) and $I$ (right, red) while the $X$ axis is space (mm). A,B subfigures are generated by J.Tica and D.Bazzoli and published in \cite{Oliver2023}}
    %TODO figure caption
    \label{fig1}
\end{figure}


A realistic mathematical description of such a biological system has non-linearities, multistability, growth and boundary conditions which are not adequately captured with linear stability analysis. The latter technique only describes the onset of pattern formation, not the final pattern.
Several studies have explored how these phenomena affect Turing pattern formation including non-linearities~\parencite{ermentrout1991stripes}, multistability~\parencite{Krause2023}, boundaries~\parencite{Arcuri1986,Maini1993, Maini1997,Krause2020, Krause2021, Woolley2022} and growth~\parencite{gaffney2010, Klika2017, Krause2019}.
However, these studies are often based on idealised domains, certain types of popular or convenient boundary conditions, and artificial growth assumptions.
Additionally, often they do not provide statistics of how Turing patterns become more or less robust to the fine-tuning problem when introducing these phenomena by conducting high-throughput parameter scans.
Therefore, a high-throughout numerical study is needed to understand how all these natural phenomena affect Turing pattern formation, in particular in a synthetic system such as the one in~\cite{Oliver2023}.

In this paper, we combine linear stability analysis and numerics, to study a simple Turing reaction-diffusion network (Fig~\ref{fig1}C) with non-linearities which leads to multistability.
Our analysis goes beyond classical Turing patterns as we also consider other types of instabilities such as Hopf and Turing-Hopf and how these can generate periodic stationary or non-stationary patterns.
We find that switching of steady states in multistable system can generate unexpected outcomes not predicted by linear stability analysis.
Additionally, growth and realistic boundary conditions are added to understand how patterning occurs in microbial colonies with synthetic Turing gene circuits. This type of model results in patterns in growing domains such as those seen in Fig~\ref{fig1}D,E which resemble the biological example in Fig~\ref{fig1}A,B. 
We find that different boundaries and growth regimes can break or form patterns, therefore adding or removing robustness to pattern formation.
Studying how all these biological phenomena affect patterning is extremely important not only in the context of engineering patterns in synthetic biology, but also to understand how robust patterns occur in developmental biology where large gene-regulatory networks commonly exhibit non-linearities and multistability, and where tissues grow while in contact with an external environments.




\section*{Results}
To study the effects of multistability, non-linearities, boundary conditions, and growth, we use a simple reaction-diffusion model which consists of a 2-node Turing topology system with non-linear Hill functions to represent gene activation and inhibition with cooperativity.
The topology can be seen in Fig~\ref{fig1}C where the activator $A$ activates itself and the inhibitor $I$, while the inhibitor $I$ inhibits the activator $A$:

\begin{subequations}
    \begin{equation}
        \pdv{A}{t}=b_{A}+V_{A} \cdot\frac{1}{1+\left(\frac{K_{AA}}{A}\right)^{n_{AA}}}\cdot\frac{1}{1+\left(\frac{I}{K_{IA}}\right)^{n_{IA}}}-\mu_{A}\cdot A + D_{A}\nabla^2 A
    \end{equation}


    \begin{equation}
        \pdv{I}{t}=\Underbrace{\textrm{basal production}}{b_{I}}+ \Underbrace{\textrm{regulated production}}{V_{I} \cdot\frac{1}{1+\left(\frac{K_{AI}}{A}\right)^{n_{AI}}}}-\Underbrace{\textrm{degradation}}{\mu_{I}\cdot I} +
        \Underbrace{\textrm{diffusion}}{D_{I}\nabla^2 I},
    \end{equation}

    \label{eq:turinghill}
\end{subequations}

with X being the concentration of species $X=A, I$ and Laplacian $\nabla^2=\partial^2/\partial x^2$ in one spatial dimension (1D). The differential equations describe the rate of change in space and time as a combination of basal production and regulated production, as well as degradation and diffusion. Parameter $b_{X}$ is the basal production rate which corresponds to the leakage of the promoter, producing molecules even when the system is fully inactivated. Furthermore, $V_{X}$ is the induced maximum production rate, $K_{XY}$ is the dissociation constant of $X$ binding to $Y$'s regulatory DNA, $n_{XY}$ is the Hill coefficient (cooperativity constant) of $X$ binding to $Y$'s regulatory DNA, and $\mu_{X}$ is the linear degradation rate of $X$. Finally, $D_{X}$ is the diffusion constant of species $X$.

The boundary conditions and domain are not strictly defined here as they vary throughout the study. The boundary conditions used are   Neumann for reflective boundaries, where the derivative at the boundary is zero, as well as Dirilichet for absorbing boundaries, where the value at the boundary is zero.
The domain used is either a fixed domain of length $L$ or a linearly, isotropically and apically growing domain, which replicates the experimental growth in 1D of the bacterial colonies in~\cite{Oliver2023} (or Fig.~\ref{fig1}A).



\subsection*{From analytical to numerical: other types of dispersion relations and patterns} \label{nogrowth}

Turing instabilities are defined using linear stability analysis (LSA) as systems which are stable without diffusion but unstable for a finite wave number as diffusion is introduced (see Methods). These Turing instabilities generally lead to stationary periodic patterns (see Fig~\ref{fig:dispersions}B). In this section, we demonstrate how sometimes the classical Turing instability theory fails to predict pattern formation.
Other types of dispersion relations beyond classical Turing instabilities can produce stationary patterns and non-stationary regular patterns that might be of interest in developmental biology.


\begin{figure}[H]
    \includegraphics[width=1\textwidth]{figures/dispersion} % The name of your image file; assumes it is in the same directory as your .tex file
    \caption{\textbf{Relationship between dispersion relation and numerical solution in mono-stable systems.} \textbf{(A-E)} Examples of dispersion relations (left) and resulting numerical solutions (right) for Stable, Turing I, Unstable, Hopf, and Turing I Hopf. \textbf{(F)} Confusion matrix linking outcomes from LSA (rows) and numerical calculation (columns). Numbers show the percentage of solutions across the LSA output rows.}
    \label{fig:dispersions} % A label for referencing this figure later in the document
\end{figure}


High-throughput studies such as \cite{Scholes2019, Zheng2016, Marcon} only consider the Turing-I type dispersion for patterning and disregard the rest.
Here, we study solutions of the classical 2-node circuit (Fig~\ref{fig1}C, Eq~\ref{eq:turinghill}) and explore beyond Turing I and stationary patterns to give a more complete view of the relationship between LSA and spatio-temporal patterns. 
First, LSA was carried out, as explained in the Methods section, for particular parameter sets to find multiple steady states with different types of stability. 
Second, numerical simulations were computed using the Crank-Nicholson algorithm, where the initial condition is drawn from a random uniform distribution around a particular steady state. By classifying dispersion-relation and pattern types, we document what type of dispersion relations in mono-stable systems can be linked to what types of patterns, to gain insights into predicting pattern formation from LSA. All the solutions used in this section are mono-stable systems to ensure a direct correlation between the dispersion relation and the pattern obtained.

All types of dispersion relations in our parameter space were classified into 5 types: stable dispersion relations have all eigenvalues $\sigma$ below zero for any wavenumber $k_{n}$ (Fig ~\ref{fig:dispersions}A). Unstable dispersion relations have a positive eigenvalue at $k_{0}=0$ which eventually drops below zero as diffusion is introduced, i.e. $k_{n}>0$ (Fig~\ref{fig:dispersions}C). Hopf-type dispersion relations, as with any unstable dispersion relation, show an instability without diffusion ($\sigma>0$ for $k_{0}=0$), which eventually drops below zero for positive wavenumbers. However, in the case of the Hopf-type dispersion relation, when the eigenvalues cross the zero line, there is a pair of complex conjugate eigenvalues (Fig~\ref{fig:dispersions}D).
A Hopf-type dispersion relation is different to a Hopf bifurcation: a bifurcation displays a shift in stability as a model parameter changes, while the Hopf-type dispersion is a change in stability as a function of the wavenumber $k_{n}$.
Turing I dispersion relations, as previously mentioned, are stable without diffusion, have an instability for a positive wavenumber, and finally become stable again for very large wavenumbers (Fig~\ref{fig:dispersions}B).
Turing I-Hopf dispersion relations, are a combination of Turing I and Hopf. As the Hopf-type dispersion, they are unstable without diffusion. Then, as $k_{n}$ is increased, the system becomes stable with a pair of complex conjugates as the eigenvalues cross the zero line.
Finally, a Turing I-type behaviour arises getting a peak above zero and decaying again for large wavenumbers (Fig~\ref{fig:dispersions}E).


Other types of dispersion relations exist, which are not displayed here such as Turing II. Unlike Turing I, the eigenvalues do not become stable again for very large wavenumbers.
Therefore, this system displays an instability at very large wavenumbers, which results in infinitesimally small wavelength patterns (at least for an infinite system). These are considered to produce spatially homogeneous solutions, except in the case of space discretization where they can produce small wavelength patterns~\parencite{Wang2022}.
However, Turing II solutions are not possible in systems such as this one where all nodes are diffusing.
This is because for $k_n \rightarrow \infty$, all eigenvalues $\sigma$ must be negative (see Eq~\ref{jacobian_diffusion}).

As with the classification of the dispersion relations, we classify the patterns produced numerically into homogeneous, temporal oscillator, non-stationary, and stationary patterns (see Methods and \nameref{sup_fig2}).
By classifying both LSA and numerical outputs, we can generate a confusion matrix with information about other types of dispersion relations and other types of spatio-temporal patterns (Fig~\ref{fig:dispersions}F). This confusion matrix shows that also other systems than Turing I can generate stationary spatial patterns. Unstable, Hopf and Turing I Hopf can produce such results too. Additionally, interesting behaviours can arise such as temporal oscillators and non-stationary patterns.


\subsection*{Feedbacks and non-linearities cause multi-stability}

Multi-stable systems are another case where linear stability analysis (LSA) fails to correctly predict pattern formation.  In this section, we demonstrate how LSA is neither sufficient 
nor necessary to predict Turing patterns in multi-stable system. In particular, we study in detail their dynamical behaviour, which can lead to the creation or breaking of patterns.
The motivation behind this arises from the high degree of multi-stability exhibited by biological systems, especially in those systems with non-linearities and feedback loops, as often occuring in biology or synthetic systems (see Eq~\ref{eq:turinghill})~\parencite{pham2020complexity, leite2009multistability}.

Using the two-node non-linear Turing topology (Eq~\ref{eq:turinghill}), multi-stable solutions were identified by finding the steady states of the system using the Newton-Raphson method. These multi-stable solutions were then studied to understand how the patterning dynamics are affected in the presence of multiple steady-state solutions.
As in the previous section, LSA was carried out to find multi-stable solutions and the Crank Nicholson algorithm was run to obtain the numerical solution for these.
Following the classical hypothesis used in the Turing literature, we only expected equations fulfilling the Turing conditions to produce patterns.
Here, we present various examples of how this hypothesis can break in the presence of multi-stability.

Fig~\ref{fig:multistability1} shows a case where diffusion-driven instability conditions are not required for Turing pattern formation.
The unstable state, having a negative dispersion relation  (Fig~\ref{fig:multistability1}C), manages to form a Turing pattern as the neighbouring Turing steady state attracts it.
It therefore produces a stationary pattern (Fig~\ref{fig:multistability1}B), even though its dispersion relation does not predict so.
The trajectory is depicted in the phase diagram (Fig~\ref{fig:multistability1}A) which shows the steady states along with the vector field to understand the potential trajectories of the system.
The phase diagram does not fully capture the dynamics as it describes the system without diffusion, while the dispersion relation and the numerical solution do consider diffusion.

\begin{figure}[H]
    \includegraphics[width=1\textwidth]{figures/multistability1}

    \caption{\textbf{Stationary patterns with multi-stability.} \textbf{(A)} Phase diagram without diffusion illustrating three distinct steady states where the derivative is zero: stable, unstable, and stable (Turing). These steady states are represented within a parameter space defined by the two concentrations for species $A$ and $I$. The vector field, indicated by light green arrows, shows the direction of the derivatives of the system at various points in the parameter space. A hand-drawn trajectory is also shown (dark green arrow), demonstrating how the unstable state may evolve into the Turing state. \textbf{(B)} Dispersion relation showing each type of state. \textbf{(C)} Numerical solutions of the three steady states with diffusion, where the unstable state unexpectedly produces a Turing-like stationary pattern. }
    \label{fig:multistability1}
\end{figure}
%TODO make font smaller in figure
%TODO On y axis you could use T (10^2) to get rid of two zeros on axis labels.

Next, we present a case where LSA incorrectly predicts stationary pattern formation.
Fig~\ref{fig:multistability2} shows an ephemeral or transient pattern that occurs in the unstable and Turing regimes.
The Turing pattern initially develops in the vicinity of the Turing steady state.
As the spatial heterogeneity is amplified and settles, it is attracted by the stable steady state, leading to the disruption of the pattern.
This type of transient pattern behavior was also recently reported in~\cite{Krause2023}.

\begin{figure}[H]
        \includegraphics[width=1\textwidth]{figures/multistability2} % The name of your image file; assumes it is in the same directory as your .tex file
    \caption{\textbf{Ephemeral patterns with multi-stability.} \textbf{(A)} Phase diagram without diffusion illustrating three distinct steady states where the derivative is zero. The hand-drawn trajectory (dark green arrow) shows an unstable state evolving into a stable (Turing) and a regular stable (non-Turing) state, indicated by thick black and green arrows, respectively. \textbf{(B)} Dispersion relation showing each type of state. \textbf{(C)} Numerical solutions of three steady states with diffusion. Unstable and Turing states produce temporary periodic stationary patterns that then disappear and become spatially homogeneous solutions.}
    \label{fig:multistability2} % A label for referencing this figure later in the document
\end{figure}

Other interesting examples can be found, for example, where an unstable state is surrounded by two Turing states. This unstable state leads robustly to a Turing pattern (Fig~\ref{sup_fig4}A).
Additionally, in some cases, the unstable system settles into Turing, but the Turing system gets pulled by the stable attractor (Fig ~\ref{sup_fig4}B). Additionally, some systems even exhibit three solutions which are homogeneous in time and space (Fig~\ref{sup_fig4}C). In this case, it would be worth investigating earlier time points with more resolution, as a pattern might appear then.
Similar interactions occur with multi-stability involving Turing I-Hopf solutions, which are mentioned in the following sections (Fig~\ref{sup_fig4}D).




\subsection*{Biological features: absorbing boundaries and growth}
As shown above, both numerical solutions and multi-stability can break the preconception that only classical Turing I systems can produce stationary periodic patterns.
Here, we look deeper into how other aspects linking the theory closer to the biological reality can also break this preconception.
In particular, we look at how adding absorbing boundaries and growth lead to a reaction-diffusion system may induce or break patterning.
This particular direction was inspired by experiments described in~\parencite{Oliver2023}, where growing bacterial colonies in agar are used as a platform to engineer Turing patterns using synthetic gene circuits (see Fig~\ref{fig1}A).

The parameter space was explored using simulations, classified as  (1) no-growth and reflective boundaries, (2) no-growth and absorbing boundaries, and finally (3) growth and absorbing boundaries. This way we can understand separately the effects of absorbing boundaries and growth in a synthetic system such as~\cite{Oliver2023}
Again, all these parameter sets have only a single steady state to ensure patterning effects are due to boundaries and growth, and not due to multi-stability.
%TODO remove ibid
When absorbing boundaries or growth are added, periodic patterns are potentially created, disrupted, or remain the same.
The Sankey diagram in Fig~\ref{fig:boundariesgrowth}A shows how this system can transition when absorbing boundaries and growth are introduced.
This transition can also be seen in Fig~\ref{sup_fig5} as a confusion matrix from reflective to absorbing boundaries and from non-growing to growing domains.
As explained in the Methods section, the classification previously used cannot be applied for absorbing boundary conditions as these prevent the pattern from being completely homogeneous or stationary.  %TODO add letter to image
Therefore, a different type of classification has to be used for reflective and absorbing boundary conditions.
For reflective boundary conditions, we classify patterns into (1) homogeneous, (2) temporal oscillators, (3) non-stationary patterns and (4) stationary patterns (see Methods and~\nameref{sup_fig2}). For absorbing boundary conditions we classify patterns into (1) homogeneous, (2) boundary effect, (3) weak pattern, (4) intermediate pattern, and (5) strong pattern (see Methods and~\nameref{sup_fig3}).
Although the classification outputs cannot be directly compared, this confusion matrix is useful for identifying interesting cases:
If absorbing boundary conditions had no effect, we would expect homogeneous and temporal oscillator categories to become no pattern, homogeneous, or boundary effect.
Additionally, we would expect stationary patterns to become strong patterns.
This is commonly the case as seen by the thicker lines in Fig~\ref{fig:boundariesgrowth}A, which shows boundaries do not often affect the pattern.
However, there are some exceptions which are further studied in Fig~\ref{fig:boundariesgrowth}B.

\begin{figure}[H]
    \includegraphics[width=1\textwidth]{figures/boundaries_growth} % The name of your image file; assumes it is in the same directory as your .tex file
    \caption{\textbf{Effects of absorbing boundaries and growth on spatio-temporal patterns.}\textbf{(A)} Sankey Diagram showing transitions across pattern classes when Absorbing boundaries and Growth is added. The thickness of the line corresponds to the amount of cases following that transition. Pink corresponds to first classification system while green corresponds to second classification system. Specific cases are highlighted in light green, blue and red. \textbf{(B)} Example of system transitioning from homogeneous (top) to strong pattern (middle) upon introduction of absorbing boundary conditions. Dispersion relation (bottom) shows a Turing instability. \textbf{(C)} Example of system transitioning from temporal oscillator (top) to strong pattern (middle) upon introduction of absorbing boundary conditions. Dispersion relation (bottom) shows a Turing-Hopf instability. \textbf{(D)} Example of system transitioning from weak pattern (top) with oscillations to a strong stationary pattern (middle) upon introduction of growing domains. Dispersion relation (bottom) shows a system without instabilities driven by diffusion.  }
    \label{fig:boundariesgrowth} % A label for referencing this figure later in the document
\end{figure}

%TODO check robert

In particular, we observe systems where an absorbing boundary condition makes a homogeneous solution become a non-stationary pattern.
Additionally, absorbing boundary conditions can turn a temporal oscillator into a stationary pattern.
Finally, by adding growth, a non-stationary pattern can be stabilized into a growing stationary pattern.

%TODO make sure you use American Englsh spelling for PLoS


\section*{Discussion}

Over the last few decades, there have been tremendous efforts in understanding how biology produces robust reproducible patterns, e.g. during embryonic development, with seminal work by Turing and others (\cite{Turing1952,Gierer1972, maini2012turing}). Robustness, sometimes called structural robustness, refers to the fraction of parameter space leading to Turing patterns. Due to the Turing conditions, this robustness is generally tiny, hindering progress in understanding developmental patterns. Here, we investigate how a highly non-linear reaction-diffusion model motivated by synthetic gene circuits can generate spatio-temporal patterns beyond predictions from linear stability analysis (LSA). The discrepancies between LSA and numerical predictions are considerable and can be attributed to non-linearities, boundary conditions, and growth. A specific focus of our work is the role of multi-stability on Turing systems, which arises from non-linearities and feedbacks loops. We describe the mechanism by which an unstable system can acquire Turing patterns, and conversely how Turing states can lose their patterns. These findings make the Turing mechanism significantly more versatile than originally assumed, and point towards investigating the effects of other biological properties such as absorbing boundary conditions or growing domains on patterning robustness. 

%TODO check structural robustness is correct
Most current robustness studies primarily focus on Turing I instabilities ~\parencite{Scholes2019, Zheng2016, Marcon}.
However, we demonstrate that numerical methods for robustness searches can uncover a wider variety of relevant spatio-temporal solutions.
For example, systems exhibiting unstable, Hopf or Turing I-Hopf dispersion relations can also produce Turing-like stationary periodic patterns (see Fig~\ref{fig:dispersions}F). Therefore, systems with such dispersion relations should not be disregarded when studying pattern formation.
Although considering these systems might slightly improve robustness, it does not fully explain the difference between robust mechanisms found in nature and the non-robust Turing patterns. The latter still require fine-tuning due to mathematical constraints.
Additionally, some systems with a Hopf-dispersion relation produce noteworthy non-stationary patterns.
These non-stationary patterns could be highly valuable in developmental and synthetic biology because arrest of gene expression could transform them into stationary patterns.
Furthermore, these non-stationary patterns might act as a pre-pattern for initial symmetry breaking, and make downstream patterning more reproducible. 
Therefore, in the context of symmetry-breaking events and patterns in developmental biology, it is crucial to consider mechanisms beyond traditional Turing dispersion relations and include Hopf, Turing I-Hopf, and simple unstable systems as potential explanations for biological patterning.


There is evidence for the above mentioned ephemeral patterns, i.e. transient patterns occurring as the system transitions from Turing to stable states, to be relevant for developmental biology. If the patterned state has a longer life time than down stream gene expression, then they suffice in temporarily activating the necessary genes to produce a permanent phenotype.
For instance, digit formation or hair follicle development requires periodic patterns only at specific times when a hormone is produced to generate fingers or hairs~\parencite{raspopovic2014digit,glover2023developmental}. 
Understanding how interactions between different steady states influence pattern formation is essential, as multi-stability is widespread and plays a critical role in biological systems~\parencite{laurent1999multistability}. In the authors' view, dynamics is more important than steady-state patterns for embryonic development~\parencite{pezzotta2023optimal}.


An important finding of our work is that absorbing boundary conditions might be part of the solution to the lack of Turing patterning robustness. A significant proportion of cases fall into the top diagonal of the confusion matrix (Fig~\ref{sup_fig5}A), suggesting that absorbing boundary conditions might enhance patterning robustness.
Specifically, absorbing boundaries can induce spatial patterning by creating non-stationary or stationary periodic patterns from spatially homogeneous patterns (Fig~\ref{fig:boundariesgrowth}B,C). Absorbing boundaries (and growth) were also used in the experimental setup in ~\parencite{Oliver2023}, which produced a high number of patterns despite the fragility of the Turing mechanism. However, adding absorbing boundaries can also disrupt spatial patterns, as shown in the Sankey diagram (Fig~\ref{fig:boundariesgrowth}A). Hence, each effect is a two-sided sword in both positvely and negatively affecting pattern formation.

In contrast to absorbing boundaries, introducing growth did not seem to improve robustness for pattern formation as more cases ended up at the bottom of the diagonal in the confusion matrix (Fig~\ref{sup_fig5}B).
This finding contradicts some literature on growth-induced Turing patterns ~\parencite{gaffney2010}.
The discrepancy may arise from differences in growth rates and types of growth (e.g. exponential or logistic).
Using varying growth rates might enhance robustness or lead to different pattern types, such as interior stripe growth instead of the outer stripe addition shown in Fig ~\ref{fig:boundariesgrowth}D-bottom~\parencite{konow2019turing}. Note that the growth rate used in this study is slower than the experimental growth rate in ~\parencite{Oliver2023} as long times were simulated to reach convergence in the non-growing reflective boundaries case. Future research may want to test robustness for specific growth rates within an experimental system. Insights into which growth rates most effectively promote pattern formation could be valuable for optimizing experiments.
Future work may also aim to develop a unified classification method that can interpret spatio-temporal patterns across different types of boundaries and both non-growing and growing domains. Such a method would produce comparable outputs to better understand the effects of growth and different boundary conditions on pattern formation.
Although limited, our approach still identifies interesting cases where growth or boundaries influence patterning.


In conclusion, by including the realistic effects of multi-stability, growth, and boundary conditions, Turing patterns begin to bridge to biological applications. In particular, absorbing boundary conditions, by setting up chemical gradients reminiscent of Wolpert's French flag model~\parencite{wolpert1969positional}, improve structural robustness of the Turing parameter space significantly. As those effects generally require the full non-linear system and hence numerical treatment, this underscores the importance of looking beyond common linear stability analysis. Applications in engineering Turing patterns with synthetic circuits are a promising way forward.


\section*{Methods}
The 2-node non-linear Turing network given by Eq~\ref{eq:turinghill} is investigated using linear stability analysis and numerical methods.
Numerically, the system is studied using non-growing domains with reflective boundary conditions, non-growing domains with absorbing boundary conditions, and finally growing domains with absorbing boundary conditions.
Numerical results are then classified into different types of patterns to understand the relationship between linear stability analysis and numerics, as well as the effects of boundary conditions and growth on patterning. The details of the numerical methods are provided here.

\subsection*{Linear stability analysis}\label{sec:lsa}
Linear stability analysis (LSA) is carried out to find out if a steady state exhibits a Turing instability (which is also called a diffusion-driven instability).
When it does, the system is capable of forming spatial patterns.
As the name describes, diffusion-driven instabilities arise in these systems when a homogeneous steady state is stable to small perturbations in the absence of diffusion, and becomes unstable in the presence of diffusion~\parencite{Glendinning1994, J.DMurray2002}.

The method of LSA, designed to check the system's stability, will be explained for a generic two morphogen reaction-diffusion system shown below:

\begin{subequations}
    \begin{equation}
        \pdv{A}{t}= f_{A}(A, I) + D_{A}\nabla^2 A
        \label{eq:RD general equation 1}
    \end{equation}
    \begin{equation}
        \pdv{I}{t} = f_{I}(A, I) + D_{I}\nabla^2 I
        \label{eq:RD general equation 2}
    \end{equation}
    \label{eq: RD general equations}
\end{subequations}
where $f_{A,I}$ are the non-linear production terms and $D_{A,I}$ are the diffusion constants of the two morphogens.


First, the steady states are defined  $A^*$ and $I^*$, which satisfy the condition:
\begin{equation}
    f_{A}(A^*,I^*)=0, \hspace{1.5cm} f_{I}(A^*,I^*)=0
\end{equation}
The non-linear reaction terms $f_{X}(A, I)$ for $X=A, I$ are then linearised around this steady state to investigate the instability to small perturbations around this steady state. Subsequently, the diffusion term $D_{X}\nabla^2 X$ is expressed as a cosine Fourier series which represents a solution with reflective boundary conditions. This results in the following expression:

\begin{subequations}
    \begin{equation}
        \pdv{(\delta A)}{t} = \pdv{f_{A}(A^*,I^*)}{A}\delta A + \pdv{f_{A}(A^*,I^*)}{I}\delta I  -D_{A}k_{n}^2\delta A
    \end{equation}
    \begin{equation}
        \pdv{(\delta I)}{t} =  \pdv{f_{I}(A^*,I^*)}{A}\delta A + \pdv{f_{I}(A^*,I^*)}{I}\delta I  -D_{I}k_{n}^2\delta I
    \end{equation}
    \label{eq:linearised RD}
\end{subequations}

The reflective boundary conditions are ensured when the derivative of diffusion is zero at the boundaries $x=[0,L]$. Note that Dirichlet boundaries are not defined here as they are not studied using linear stability analysis, only numerically. Therefore due to the constraints at the reflective boundaries, $k_{n}$ must be
\newcommand{\nat}{\numberset{N}}
\newcommand{\numberset}[1]{\mathbb{#1}}

\begin{equation}
    k_{n}=\frac{n \pi}{L} \hspace{10pt} \forall \hspace{5pt} {n \in \nat }
    \label{kn}
\end{equation}

In this case, we are interested on the growth or decay of the perturbations over time. The stability of this system can be tested by calculating the eigenvalues $\sigma$ of its Jacobian where matrix elements are evaluated at steady state
\begin{equation}
    J = \begin{bmatrix}
            \pdv{f_{A}}{A} - D_{A}k_{n}^2 &
            \pdv{f_{A}}{I}  \\
            \pdv{f_{I}}{A} &
            \pdv{f_{I}}{I} - D_{I}k_{n}^2
    \end{bmatrix}
    \label{jacobian_diffusion}
\end{equation}


\begin{itemize}
    \item If $\sigma > 0$: perturbation ($\delta X$) grows making $\pdv{\delta X}{t} > 0$.
    Therefore, the steady state is unstable.
    \item If $\sigma < 0$: perturbation ($\delta X$) decays making $\pdv{\delta X}{t} < 0$.
    Therefore, the steady state is stable,
\end{itemize}

with $\sigma=0$ the marginal stable case, which generally requires further investigation, but numerically is easily avoided.

\subsubsection*{Implementation of linear stability analysis}
The steady states of the system are found using the Newton-Raphson algorithm with 100 initial conditions and a tolerance value of $10^{-6}$. The stability of these steady states is analysed without diffusion by setting $k_{n}=0$. If any of the eigenvalues have a real positive part, the steady state is unstable without diffusion. The stability of the steady state is analysed by solving for the eigenvalues of the Jacobian for all $k_{n}=\frac{n\pi}{L} \hspace{0.1cm}\forall \{n \in \mathbb{N} : n \leq 5,000\} $, meaning 5,000 $k$'s are sampled to estimate the dispersion relation. Similarly, if any of the eigenvalues have a real positive part, the steady state is unstable with diffusion. LSA for a single steady state takes approximately approximately 0.5s.
%TODO : Remove or explain on what computer system, and also refer to speed of calculation at other places of methods.

\subsection*{Numerical methods}\label{numerical methods}
A finite difference method is chosen to solve the system of non-linear PDEs. By discretizing time and space, the two independent variables can be expressed as:
\begin{subequations}
    \begin{equation}
        t_{n} = n \cdot \Delta t, \quad n=0,\dots,N-1
    \end{equation}
    \begin{equation}
        x_{j} = j \cdot \Delta x, \quad j=0, \dots,J-1,
    \end{equation}
\end{subequations}
with $N$ and $J$  the number of discrete time and space points in our grid, respectively. Furthermore, $\Delta t$ and $\Delta x$ are the time steps and the space steps correspondingly, with $T$ and $L$  their final time and space values. The aim is to derive a numerical solution that approximates the unknown analytical solution, $U(j\Delta x, n\Delta t)\approx u( j\Delta x, n\Delta t)$, where $U$ is the analytical solution and $u$ is the numerical solution.

The Crank-Nicolson numerical scheme is chosen as it is unconditionally stable as shown by von Neumann stability analysis~\parencite{strikwerda2004finite}.
The unconditional stability is important to allow for larger $\Delta t$ and $\Delta x$, without getting an amplification of errors. Larger $\Delta t$ and $\Delta x$ will result in reduced computational cost.


\subsubsection*{Crank Nicolson method}\label{cranknicolson}
Consider a reaction-diffusion system with one spatial dimension
\begin{equation}
    \frac{\partial u}{\partial t} =  f(u) + D\pdvn{2}{u}{x}
\end{equation}
The spatial part of the equation can be approximated by
\begin{equation}
    \pdvn{2}{u}{x} \biggr\rvert_{x=j\Delta x,t=n\Delta t} \approx \frac{1}{2\Delta x^{2}}\left( U^{n}_{j+1} -  2U^{n}_{j} + U^{n}_{j-1} + U^{n+1}_{j+1} - 2U^{n+1}_{j} + U^{n+1}_{j-1}\right),
\end{equation}
while the production function can be approximated to $f ( U^{n}_{j})$. The approximations can be better visualized using the CN stencil (see Fig~\ref{sup_fig1})
Applying the CN stencil to the grid point $(i, j)$, the reaction-diffusion system can be expressed as

\begin{equation}
    \frac{U^{n+1}_{j} - U^{n}_{j}}{\Delta t} = \frac{D}{2\Delta x^{2}}\left( U^{n}_{j+1} -  2U^{n}_{j} + U^{n}_{j-1} + U^{n+1}_{j+1} - 2U^{n+1}_{j} + U^{n+1}_{j-1}\right) +  f( U^{n}_{j})
    \label{CN_stencil}
\end{equation}
By reordering this approximation into a set of linear equations, the resulting problem is defined by matrices $\bf A$ and $\bf B$, where $\textbf{U}^{n+1} = [U^{n}_{0}, \ldots , U^{n}_{J-1}]$. The simplified system can be expressed as:
\begin{equation}
    \textbf{U}^{n+1} = A^{-1}(B\textbf{U}^{n} + f^{n})
\end{equation}
This method simplifies the complex system into a linear system that can be solved numerically. The returned solution is a 1D solution of the corresponding reaction-diffusion system. Although the method is unconditionally stable, the solution can contain oscillations if $ \frac{D\Delta t}{\Delta x^{2}} >\frac{1}{2} $ \parencite{trefethen1996finite}. Therefore, the ratio will be kept below $\frac{1}{2}$ to avoid errors.

\subsubsection*{Defining boundary conditions}\label{methods_boundary_conditions_CN}
The CN method can be implemented with Neumann reflective boundary conditions or Dirichlet absorbing boundary conditions.
For Neumann reflective boundary conditions:
\begin{equation}
    \frac{\partial u}{\partial t} =  f(u) + D\pdvn{2}{u}{x},   \quad \quad \quad \quad \quad \quad \pdv{u}{x}\biggr\rvert_{x=0,L}=0,
\end{equation}

where the values of $U_{j}$ at $j=0$ and $j=J-1$ are

\begin{equation}
    U_{j=0} = U_{j=-1}  \quad \quad \&  \quad \quad  U_{j=J-1} = U_{j=J},
\end{equation}

which are placed into the CN stencil shown in Eq~\ref{CN_stencil}.
These values are chosen to ensure the derivative at the boundary is zero.

Similarly, Dirichlet absorbing boundary conditions are represented by the following system

\begin{equation}
    \frac{\partial u}{\partial t} =  f(u) + D\pdvn{2}{u}{x},   \quad \quad \quad \quad \quad \quad U\biggr\rvert_{x=0,L}=0
\end{equation}

and have values of $U_{j}$ at $j=-1$ and $j=J$ such as

\begin{equation}
    U_{j=-1} = 0  \quad \quad \&  \quad \quad  U_{j=J} = 0,
\end{equation}

which are placed into the CN stencil shown in Eq~\ref{CN_stencil},
ensuring the value at the boundary is zero.


\subsubsection*{Defining growing domains}
Growth is introduced as apical isotropic linear growth, where cells are added to both boundaries with a linear growth rate.
Linear growth is chosen as it is a good approximation to the growth observed in experimental bacterial colonies in Fig~\ref{fig1}A~\parencite{Oliver2023}.
In these colonies, cell division occurs mainly at the edges Hence, we assume only edge division to simplify the model.
To further reduce the model and computational power required, solutions are studied in 1D.
Growth of the tissue is encoded in a 1D binary vector, where cells are denoted as 1 and empty space as 0.
The number of 1's grows linearly, which represents the expanding tissue.
This vector is used as a mask, where 1's determine the computation of reaction-diffusion terms and 0's determine only the computation of diffusion.


\subsection*{Classification methods}\label{classification}

\subsubsection*{Reflective boundary classification}\label{numerical_classification1}
We first develop a method to classify the patterns produced numerically into (1) homogeneous, (2) temporal oscillator, as well as (3) non-stationary and (4) stationary patterns.
We use a decision tree for the classification where the two layers are spatial homogeneity and convergence in time.
This decision tree leads to the 4 types of patterns mentioned.

A pattern will be considered spatially homogeneous if the final snapshot $U$ for any of the two molecular species fulfils the following condition
\begin{equation}
    \frac{max(U) - min(U)}{max(U)} \leq 0.01
\end{equation}
A pattern will be considered converged if the last 30 time points for any of the two molecular species fulfils the following condition
\begin{equation}
    \frac{max(U[-30:]) - min(U[-30:])}{max(U[-30:])} \leq 0.05,
\end{equation}
using common pseudocode splice notation for numerical arrays. The thresholds chosen were fine-tuned by testing them on the numerical patterns to obtain the best classification results.
Using these two characteristics, spatial homogeneity and convergence, we can obtain 4 classes of patterns as seen in Fig~\ref{sup_fig2}:
\begin{enumerate}
    \item Homogeneous patterns are homogeneous in space and converge in time.
    \item Temporal oscillators, also called limit cycles, are homogeneous in space but do not converge, as they oscillate in time.
    \item Non-stationary patterns are not homogeneous in space and do not converge in time.
    \item Finally, stationary patterns are not homogeneous in space but converge in time.
\end{enumerate}
\subsubsection*{Absorbing boundary classification}\label{numerical_classification2}
With absorbing boundary conditions and growth, patterns are rarely spatially homogeneous or converged, which makes the previous classification method unsuitable. Hence, a new classification system is developed based on the number of peaks.
Peaks are detected using the Python $find\_peaks$ algorithm with parameter $prominence=0.05$.
Again, the thresholds for the peak finding algorithm need to be fine-tuned to avoid misclassification.

The peak classification shown in \ref{sup_fig3}, retrieves information on whether there is a pattern at all and whether this pattern is only a pattern at the boundary or a periodic pattern that scales with tissue length with constant wave length as Turing patterns do.
The 5 different types of patterns are:
\begin{enumerate}
    \item Patterns with one peak are considered homogeneous as they result from the morphogens being reduced at the boundary due to absorption (\nameref{sup_fig3}, no pattern, homogeneous).
    \item Patterns with two peaks are also considered not to be patterned states as the two peaks might arise at the boundary for one of the diffusors due to the depletion of the other (\nameref{sup_fig3}, no pattern, boundary effect).
    \item Patterns with three peaks start displaying a pattern more similar to Turing repeats, although we cannot prove the number of peaks would scale with tissue length  (\nameref{sup_fig3}, weak pattern).
    \item Patterns with four peaks could still be purely a boundary effect, but is less likely as the number of repeats points towards tissue scaling being possible (\nameref{sup_fig3}, intermediate pattern).
    \item Finally, patterns with five peaks and above are considered strong patterns and would be most similar to classical Turing patterns in non-growing reflective boundary domains (\nameref{sup_fig3}, strong pattern).
\end{enumerate}





\section*{Supporting information}
%
%% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
\paragraph*{Fig S1}
\label{S1_Fig}
{\bf Crank-Nicolson (CN) algorithm for numerical solution.}  Geometric representation (stencil) with nodes and edges that represents the points of interest for the numerical approximation. The points of interest, which are the ones present in the equations, are shown in green. Labels $j$ and $n$ are the current space and time points. The CN stencil has one spatial dimension and one temporal dimension, with axes labels time and space ($x$).


    
\paragraph*{Fig S2}
\label{sup_fig2}
{\bf Decision tree for pattern classification in non-growing domains with reflective boundaries}. The decision tree is based on two layers: spatial homogeneity and convergence. The numerical solutions for the four different pattern outcomes including  homogeneous, temporal oscillator, non-stationary, and stationary pattern as shown below. In the four numerical solutions, time is in the $y$ axis, space in the $x$ axis and concentration in the colorbar.

\paragraph*{Fig S3}
\label{sup_fig3}
{\bf Pattern classification with absorbing boundaries.} Patterns are classified according to the number of peaks as shown in the 5 cases of this figure.  %TODO write S3


\paragraph*{Fig S4}
\label{sup_fig4}
{\bf Other types of multistability dynamics.} \textbf{(A)} Unstable state converges into Turing. \textbf{(B)} Unstable state produces pattern, while Turing state loses pattern. \textbf{(C)} Multistability disrupts all patterns. \textbf{(D)} Turing I-Hopf state attracts unstable state and generates pattern.


\paragraph*{Fig S5}
\label{sup_fig5}
{\bf Confusion matrix for pattern transitions.} \textbf{(A)} Numerical outcome of reflective boundaries ($y$ axis) versus absorbing boundaries ($x$ axis).\textbf{(B)} Numerical outcome of non-growing domains ($y$ axis) versus growing-domains ($x$ axis). Numbers show the percentage of solutions across the row.


\section*{Acknowledgments}
We gratefully acknowledge the support and funding provided by the EPSRC CDT BioDesign Engineering. We would also like to extend our sincere thanks to Prof. Mark Isalan for his invaluable guidance on the experimental aspects as well as Dr. Jure Tica and Dr. Dario Bazzoli for providing experimental images in Fig~\ref{fig1}A,B. Additionally, we thank our colleagues, Dr. Roozbeh H. Pazuki
 and Antonio Matas Gil, for their insightful discussions which contributed to this work.

\printbibliography[heading=bibintoc]

%
%\nolinenumbers
%
%% Either type in your references using
%% \begin{thebibliography}{}
%% \bibitem{}
%% Text
%% \end{thebibliography}
%%
%% or
%%
%% Compile your BiBTeX database using our plos2015.bst
%% style file and paste the contents of your .bbl file
%% here. See http://journals.plos.org/plosone/s/latex for
%% step-by-step instructions.
%%
%\begin{thebibliography}{10}
%
%\bibitem{bib1}
%Conant GC, Wolfe KH.
%\newblock {{T}urning a hobby into a job: how duplicated genes find new
%  functions}.
%\newblock Nat Rev Genet. 2008 Dec;9(12):938--950.
%
%\bibitem{bib2}
%Ohno S.
%\newblock Evolution by gene duplication.
%\newblock London: George Alien \& Unwin Ltd. Berlin, Heidelberg and New York:
%  Springer-Verlag.; 1970.
%
%\bibitem{bib3}
%Magwire MM, Bayer F, Webster CL, Cao C, Jiggins FM.
%\newblock {{S}uccessive increases in the resistance of {D}rosophila to viral
%  infection through a transposon insertion followed by a {D}uplication}.
%\newblock PLoS Genet. 2011 Oct;7(10):e1002337.

%\end{thebibliography}

\printbibliography[heading=bibintoc]


\end{document}

